### PR TITLE
pb3206: Not returning error if the storageclass is missing on destination cluster.

### PR DESCRIPTION
**Pull request by siva-portworx**
**What type of PR is this?**
>bug

**What this PR does / why we need it**:
```
pb3206: Not returning error if the storageclass is missing on destination cluster
```

**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:
No, will be covered with pb-3174

**Does this change need to be cherry-picked to a release branch?**:
yes. 2.12

Testing
Started the failing jenkins job with the  fixed image:
https://jenkins.pwx.dev.purestorage.com/job/Stork/view/Stork%202.12-dev%20%20C7/job/2.12-backup-k8s-1-23-0/376/console

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -2621,67 +2621,69 @@ func (p *portworx) UpdateMigratedPersistentVolumeSpec(
 	if len(pv.Spec.StorageClassName) != 0 {
 		sc, err := storage.Instance().GetStorageClass(pv.Spec.StorageClassName)
 		if err != nil {
-			return nil, fmt.Errorf("failed in getting the storage class [%v]: %v", pv.Spec.StorageClassName, err)
+			logrus.Warnf("failed in getting the storage class [%v]: %v", pv.Spec.StorageClassName, err)
```
Comment (prashanthpx) [inline drivers/volume/portworx/portworx.go:2624]: Don't we need to return error from here ?

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -2621,67 +2621,69 @@ func (p *portworx) UpdateMigratedPersistentVolumeSpec(
...
 				}
-				if val == templatizedNamespace {
-					pv.Spec.CSI.ControllerExpandSecretRef.Namespace = dstNamespace
-				} else {
-					pv.Spec.CSI.ControllerExpandSecretRef.Namespace = val
+				if val, ok := sc.Parameters[controllerExpandSecretNamespace]; ok {
+					if pv.Spec.CSI.ControllerExpandSecretRef == nil {
+						pv.Spec.CSI.ControllerExpandSecretRef = &v1.SecretReference{}
+					}
+					if val == templatizedNamespace {
```
Comment (prashanthpx) [inline drivers/volume/portworx/portworx.go:2653]: Here asumption is `templatizedNamespace="${pvc.name}"` but if a user chooses something else as this is user choice, one suggestion is to have a Regex here to check for anything starting with `$` and use the val as is. 
When i was searching for this on csi website, i found an example https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html#per-volume-secrets where it is different.

**Review by prashanthpx — APPROVED** [review]

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (siva-portworx) [reply drivers/volume/portworx/portworx.go:2624]: No, if the storageclass is not present, we will return error. In that case, we will not add the CSI section. That is the fix

File: drivers/volume/portworx/portworx.go (hunk 2 above)
Comment (siva-portworx) [reply drivers/volume/portworx/portworx.go:2653]: Good point. As per our document, we are suggesting this. 
https://docs.portworx.com/operations/operate-kubernetes/storage-operations/csi/security/
To support $(pvc.Name)-xxxxxxx, we need to change in other places as well. We can take it as separate fix.

**Review by adityadani — APPROVED** [review]

**Pull request merged**